pkg/ottl/ottlfuncs/README.md

```diff
@@ -338,10 +338,16 @@ The `replace_all_patterns` function replaces any segments in a string value or k
 
 If one or more sections of `target` match `regex` they will get replaced with `replacement`.
 
+The `replacement` string can refer to matched groups using [regexp.Expand syntax](https://pkg.go.dev/regexp#Regexp.Expand).
+
 Examples:
 
 - `replace_all_patterns(attributes, "value", "/account/\\d{4}", "/account/{accountId}")`
 - `replace_all_patterns(attributes, "key", "/account/\\d{4}", "/account/{accountId}")`
+- `replace_all_patterns(attributes, "key", "^kube_([0-9A-Za-z]+_)", "k8s.$$1.")`
+
+Note that when using OTTL within the collector's configuration file, `$` must be escaped to `$$` to bypass
+environment variable substitution logic. If using OTTL outside of collector configuration, `$` should not be escaped.
```

Comment:
```suggestion
Note that when using OTTL within the collector's configuration file, `$` must be escaped to `$$` to bypass
environment variable substitution logic. To input a literal `$` from the configuration file, use `$$$`. If using OTTL outside of collector configuration, `$` should not be escaped and a literal `$` can be entered using `$$`.
```
I realize this will likely be a rare case, but I think giving users a concrete reference would be helpful since we're dealing with multiple levels of escaping.

I tried to explain it as directly as possible here, but feel free to reword it if it's still unclear.